### PR TITLE
DEVOPS-2787 update drone pipeline

### PR DESCRIPTION
**Pull request by redondos**
Changes:
- new `make dist` target
- added LICENSE
- pipeline steps for:
  - s3 cache
  - snyk
  - github releases
  - license check

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 GO_MODULE := $(shell git config --get remote.origin.url | grep -o 'github\.com[:/][^.]*' | tr ':' '/')
-CMD_NAME := $(shell basename ${GO_MODULE})
+CMD_NAME := kanopy-codegen
+VERSION := $(shell grep '^const Version =' internal/version/version.go | cut -d\" -f2)
```
Comment (dskatz) [inline Makefile:3]: nit: I'll add some official documentation to our styleguide to use ldflags to source the version into the build process

**Review by dskatz — COMMENTED** [review]
This is my bad, I forgot to specify adding a license-check step similar to the [cdnvalidator](https://github.com/kanopy-platform/cdnvalidator) since this is an open source repo.

File: Makefile (hunk 1 above)
Comment (colinhoglund) [reply Makefile:3]: Should this instead be included in https://github.com/10gen/kanopy-app-go?

File: internal/version/version.go [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package version
+
+// Version is the current version
+const Version = "0.0.1"
```
Comment (colinhoglund) [inline internal/version/version.go:4]: semver should start at `0.1.0`

File: Makefile (hunk 1 above)
Comment (dskatz) [reply Makefile:3]: Yes.

File: .drone.yml [hunk 3]
```diff
@@ -4,20 +4,78 @@ type: kubernetes
...
+        memory: 512MiB
+
+  - name: license-check
+    image: public.ecr.aws/kanopy/licensed-go:3.4.4
+    commands:
+      - licensed cache
+      - licensed status
+
+  - name: snyk
+    image: snyk/snyk:golang-1.14
```
Comment (dskatz) [inline .drone.yml:41]:
```suggestion
    image: snyk/snyk:golang-1.16
```

**Review by dskatz — APPROVED** [review]

**Pull request merged**